### ISSUE TITLE
W1011 false positive

### ISSUE THREAD

**Issue by gabelton** [aws-cloudformation/cfn-lint#3100]
### CloudFormation Lint Version

0.80.2

### What operating system are you using?

Mac and ubuntu via github action

### Describe the bug

I'm seeing W1011 rule violation because of missing keys in a mapping, but I think this is a false positive because I'm providing a `DefaultValue`, so it shouldn't matter that the key is missing.

### Expected behavior

I expect not to see this rule violation when I run `cfn-lint`

### Reproduction template

```
Mappings:
  dwS3BucketEnvironmentConfigMap:
    # Create an entry for each environment

    prod:
      BucketName: 'digital-workspace-v2-prod'
      DeletionPolicy: Delete
    hotfix:
      BucketName: 'digital-workspace-v2-hotfix'
      DeletionPolicy: Delete
    staging:
      BucketName: 'digital-workspace-v2-staging'
      DeletionPolicy: Delete
    training:
      BucketName: 'digital-workspace-v2-training'
      DeletionPolicy: Delete
    dev:
      BucketName: 'digital-workspace-v2-dev'
      DeletionPolicy: Delete

Conditions:
  dwS3BucketApplyRetentionPolicy: !Not
    - !Equals
      - !FindInMap
        - dwS3BucketEnvironmentConfigMap
        - !Ref Env
        - RetentionMode
        - DefaultValue: No retention
      - No retention
  dwS3BucketRetentionPolicyDurationIsInDays: !Equals
    - !FindInMap
      - dwS3BucketEnvironmentConfigMap
      - !Ref Env
      - RetentionDurationType
      - DefaultValue: Days
    - Days
```

EDIT: I'm seeing the same after upgrading to 0.86.0 and running locally on mac m1

**Comment by kddejong** [comment]
Did you add the Transform `AWS::LanguageExtensions`?

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-findinmap-enhancements.html

**Comment by kddejong** [comment]
Fixed and merged.  Closing and we will release today or tomorrow.

**Comment by gabelton** [comment]
@kddejong thank you for looking into this, but I think it's still cropping up in 0.87.0

I'm running against this template

```
# Generated by platform-helper 6.1.0 / 2024-03-14 15:33:40
Transform: AWS::LanguageExtensions

Parameters:
  # Copilot required Parameters...
  App:
    Type: String
    Description: Your application's name.
  Env:
    Type: String
    Description: The environment name your service, job, or workflow is being deployed to.

Mappings:
  demodjangoS3BucketEnvironmentConfigMap:
    # Create an entry for each environment

    ant:
      BucketName: 'demodjango-s3-bucket-ant'
      DeletionPolicy: Delete
    toolspr:
      BucketName: 'demodjango-s3-bucket-toolspr'
      DeletionPolicy: Delete
    staging:
      BucketName: 'demodjango-s3-bucket-staging'
      DeletionPolicy: Delete
    jayesh:
      BucketName: 'demodjango-s3-bucket-jayesh'
      DeletionPolicy: Delete
    willg:
      BucketName: 'demodjango-s3-bucket-willg'
      DeletionPolicy: Delete
    dev:
      BucketName: 'demodjango-s3-bucket-dev'
      DeletionPolicy: Delete
    john:
      BucketName: 'demodjango-s3-bucket-john'
      DeletionPolicy: Delete
    yusuf:
      BucketName: 'demodjango-s3-bucket-yusuf'
      DeletionPolicy: Delete
    samm:
      BucketName: 'demodjango-s3-bucket-samm'
      DeletionPolicy: Delete
    gabe:
      BucketName: 'demodjango-s3-bucket-gabe'
      DeletionPolicy: Delete

Conditions:
  demodjangoS3BucketApplyRetentionPolicy: !Not
    - !Equals
      - !FindInMap
        - demodjangoS3BucketEnvironmentConfigMap
        - !Ref Env
        - RetentionMode
        - DefaultValue: No retention
      - No retention
  demodjangoS3BucketRetentionPolicyDurationIsInDays: !Equals
    - !FindInMap
      - demodjangoS3BucketEnvironmentConfigMap
      - !Ref Env
      - RetentionDurationType
      - DefaultValue: Days
    - Days

Resources:
  demodjangoS3BucketKMSKey:
    Type: "AWS::KMS::Key"
    DeletionPolicy: !FindInMap [demodjangoS3BucketEnvironmentConfigMap, !Ref Env, DeletionPolicy]
    UpdateReplacePolicy: Retain
    Properties:
      Description: "KMS Key for RDS encryption"
      KeyPolicy:
        Version: '2012-10-17'
        Id: !Sub '${App}-${Env}-demodjangoS3Bucket-key'
        Statement:
        - Sid: Enable IAM User Permissions
          Effect: Allow
          Principal:
            AWS: !Sub "arn:aws:iam::${AWS::AccountId}:root"
          Action: kms:*
          Resource: '*'

  demodjangoS3BucketKeyAlias:
    Type: 'AWS::KMS::Alias'
    DeletionPolicy: !FindInMap [demodjangoS3BucketEnvironmentConfigMap, !Ref Env, DeletionPolicy]
    UpdateReplacePolicy: Retain
    Properties:
      AliasName: !Sub 'alias/${App}-${Env}-demodjangoS3Bucket-key'
      TargetKeyId: !Ref demodjangoS3BucketKMSKey

  demodjangoS3BucketBucket:
    Metadata:
      'aws:copilot:description': 'An Amazon S3 bucket to store and retrieve objects for demodjangoS3Bucket'
    Type: AWS::S3::Bucket
    DeletionPolicy: !FindInMap [demodjangoS3BucketEnvironmentConfigMap, !Ref Env, DeletionPolicy]
    UpdateReplacePolicy: Retain
    Properties:
      BucketName: !FindInMap [demodjangoS3BucketEnvironmentConfigMap, !Ref Env, BucketName]
      AccessControl: Private
      BucketEncryption:
        ServerSideEncryptionConfiguration:
        - ServerSideEncryptionByDefault:
            SSEAlgorithm: aws:kms
            KMSMasterKeyID: !Ref demodjangoS3BucketKMSKey
      ObjectLockConfiguration:
        !If
        - demodjangoS3BucketApplyRetentionPolicy
        - ObjectLockEnabled: Enabled
          Rule:
            DefaultRetention: !If
              - demodjangoS3BucketRetentionPolicyDurationIsInDays
              - Mode: !FindInMap [demodjangoS3BucketEnvironmentConfigMap, !Ref Env, RetentionMode]
                Days: !FindInMap
                  - demodjangoS3BucketEnvironmentConfigMap
                  - !Ref Env
                  - RetentionDuration
              - Mode: !FindInMap [demodjangoS3BucketEnvironmentConfigMap, !Ref Env, RetentionMode]
                Years: !FindInMap
                  - demodjangoS3BucketEnvironmentConfigMap
                  - !Ref Env
                  - RetentionDuration
        - !Ref AWS::NoValue
      VersioningConfiguration:
        Status: Enabled
      PublicAccessBlockConfiguration:
        BlockPublicAcls: true
        BlockPublicPolicy: true
        IgnorePublicAcls: true
        RestrictPublicBuckets: true
      Tags:
        - Key: Copilot-application
          Value: !Ref App
        - Key: Copilot-environment
          Value: !Ref Env
        - Key: Copilot-deletion-policy
          Value: !FindInMap [demodjangoS3BucketEnvironmentConfigMap, !Ref Env, DeletionPolicy]

  demodjangoS3BucketBucketPolicy:
    Metadata:
      'aws:copilot:description': 'A bucket policy to deny unencrypted access to the bucket and its contents'
    Type: AWS::S3::BucketPolicy
    DeletionPolicy: !FindInMap [demodjangoS3BucketEnvironmentConfigMap, !Ref Env, DeletionPolicy]
    UpdateReplacePolicy: Retain
    Properties:
      PolicyDocument:
        Version: '2012-10-17'
        Statement:
          - Sid: ForceHTTPS
            Effect: Deny
            Principal: '*'
            Action: 's3:*'
            Resource:
              - !Sub ${ demodjangoS3BucketBucket.Arn}/*
              - !Sub ${ demodjangoS3BucketBucket.Arn}
            Condition:
              Bool:
                "aws:SecureTransport": false
      Bucket: !Ref demodjangoS3BucketBucket



Outputs:
  demodjangoS3BucketName:
    Description: "The name of a user-defined bucket."
    Value: !Ref demodjangoS3BucketBucket

  demodjangoS3BucketKMSKeyARN:
    Description: "ARN of the KMS Key"
    Value: !GetAtt demodjangoS3BucketKMSKey.Arn
    Export:
      Name: !Sub "${App}-${Env}-demodjangoS3Bucket-KMSKeyARN"
```

and getting the following errors:

```
[W1011 FindInMap second key "RetentionMode" doesn't exist in map "demodjangoS3BucketEnvironmentConfigMap" under "ant" at Resources/demodjangoS3BucketBucket/Properties/ObjectLockConfiguration/Fn::If/1/Rule/DefaultRetention/Fn::If/1/Mode/Fn::FindInMap
copilot/environments/addons/demodjango-s3-bucket.yml:112:85

W1011 FindInMap second key "RetentionMode" doesn't exist in map "demodjangoS3BucketEnvironmentConfigMap" under "toolspr" at Resources/demodjangoS3BucketBucket/Properties/ObjectLockConfiguration/Fn::If/1/Rule/DefaultRetention/Fn::If/1/Mode/Fn::FindInMap
copilot/environments/addons/demodjango-s3-bucket.yml:112:85

W1011 FindInMap second key "RetentionMode" doesn't exist in map "demodjangoS3BucketEnvironmentConfigMap" under "staging" at Resources/demodjangoS3BucketBucket/Properties/ObjectLockConfiguration/Fn::If/1/Rule/DefaultRetention/Fn::If/1/Mode/Fn::FindInMap
copilot/environments/addons/demodjango-s3-bucket.yml:112:85

W1011 FindInMap second key "RetentionMode" doesn't exist in map "demodjangoS3BucketEnvironmentConfigMap" under "jayesh" at Resources/demodjangoS3BucketBucket/Properties/ObjectLockConfiguration/Fn::If/1/Rule/DefaultRetention/Fn::If/1/Mode/Fn::FindInMap
copilot/environments/addons/demodjango-s3-bucket.yml:112:85

W1011 FindInMap second key "RetentionMode" doesn't exist in map "demodjangoS3BucketEnvironmentConfigMap" under "willg" at Resources/demodjangoS3BucketBucket/Properties/ObjectLockConfiguration/Fn::If/1/Rule/DefaultRetention/Fn::If/1/Mode/Fn::FindInMap
copilot/environments/addons/demodjango-s3-bucket.yml:112:85

W1011 FindInMap second key "RetentionMode" doesn't exist in map "demodjangoS3BucketEnvironmentConfigMap" under "dev" at Resources/demodjangoS3BucketBucket/Properties/ObjectLockConfiguration/Fn::If/1/Rule/DefaultRetention/Fn::If/1/Mode/Fn::FindInMap
copilot/environments/addons/demodjango-s3-bucket.yml:112:85

W1011 FindInMap second key "RetentionMode" doesn't exist in map "demodjangoS3BucketEnvironmentConfigMap" under "john" at Resources/demodjangoS3BucketBucket/Properties/ObjectLockConfiguration/Fn::If/1/Rule/DefaultRetention/Fn::If/1/Mode/Fn::FindInMap
copilot/environments/addons/demodjango-s3-bucket.yml:112:85

W1011 FindInMap second key "RetentionMode" doesn't exist in map "demodjangoS3BucketEnvironmentConfigMap" under "yusuf" at Resources/demodjangoS3BucketBucket/Properties/ObjectLockConfiguration/Fn::If/1/Rule/DefaultRetention/Fn::If/1/Mode/Fn::FindInMap
copilot/environments/addons/demodjango-s3-bucket.yml:112:85

W1011 FindInMap second key "RetentionMode" doesn't exist in map "demodjangoS3BucketEnvironmentConfigMap" under "samm" at Resources/demodjangoS3BucketBucket/Properties/ObjectLockConfiguration/Fn::If/1/Rule/DefaultRetention/Fn::If/1/Mode/Fn::FindInMap
copilot/environments/addons/demodjango-s3-bucket.yml:112:85

W1011 FindInMap second key "RetentionMode" doesn't exist in map "demodjangoS3BucketEnvironmentConfigMap" under "gabe" at Resources/demodjangoS3BucketBucket/Properties/ObjectLockConfiguration/Fn::If/1/Rule/DefaultRetention/Fn::If/1/Mode/Fn::FindInMap
copilot/environments/addons/demodjango-s3-bucket.yml:112:85

W1011 FindInMap second key "RetentionDuration" doesn't exist in map "demodjangoS3BucketEnvironmentConfigMap" under "ant" at Resources/demodjangoS3BucketBucket/Properties/ObjectLockConfiguration/Fn::If/1/Rule/DefaultRetention/Fn::If/1/Days/Fn::FindInMap
copilot/environments/addons/demodjango-s3-bucket.yml:116:21

W1011 FindInMap second key "RetentionDuration" doesn't exist in map "demodjangoS3BucketEnvironmentConfigMap" under "toolspr" at Resources/demodjangoS3BucketBucket/Properties/ObjectLockConfiguration/Fn::If/1/Rule/DefaultRetention/Fn::If/1/Days/Fn::FindInMap
copilot/environments/addons/demodjango-s3-bucket.yml:116:21

W1011 FindInMap second key "RetentionDuration" doesn't exist in map "demodjangoS3BucketEnvironmentConfigMap" under "staging" at Resources/demodjangoS3BucketBucket/Properties/ObjectLockConfiguration/Fn::If/1/Rule/DefaultRetention/Fn::If/1/Days/Fn::FindInMap
copilot/environments/addons/demodjango-s3-bucket.yml:116:21

W1011 FindInMap second key "RetentionDuration" doesn't exist in map "demodjangoS3BucketEnvironmentConfigMap" under "jayesh" at Resources/demodjangoS3BucketBucket/Properties/ObjectLockConfiguration/Fn::If/1/Rule/DefaultRetention/Fn::If/1/Days/Fn::FindInMap
copilot/environments/addons/demodjango-s3-bucket.yml:116:21

W1011 FindInMap second key "RetentionDuration" doesn't exist in map "demodjangoS3BucketEnvironmentConfigMap" under "willg" at Resources/demodjangoS3BucketBucket/Properties/ObjectLockConfiguration/Fn::If/1/Rule/DefaultRetention/Fn::If/1/Days/Fn::FindInMap
copilot/environments/addons/demodjango-s3-bucket.yml:116:21

W1011 FindInMap second key "RetentionDuration" doesn't exist in map "demodjangoS3BucketEnvironmentConfigMap" under "dev" at Resources/demodjangoS3BucketBucket/Properties/ObjectLockConfiguration/Fn::If/1/Rule/DefaultRetention/Fn::If/1/Days/Fn::FindInMap
copilot/environments/addons/demodjango-s3-bucket.yml:116:21

W1011 FindInMap second key "RetentionDuration" doesn't exist in map "demodjangoS3BucketEnvironmentConfigMap" under "john" at Resources/demodjangoS3BucketBucket/Properties/ObjectLockConfiguration/Fn::If/1/Rule/DefaultRetention/Fn::If/1/Days/Fn::FindInMap
copilot/environments/addons/demodjango-s3-bucket.yml:116:21

W1011 FindInMap second key "RetentionDuration" doesn't exist in map "demodjangoS3BucketEnvironmentConfigMap" under "yusuf" at Resources/demodjangoS3BucketBucket/Properties/ObjectLockConfiguration/Fn::If/1/Rule/DefaultRetention/Fn::If/1/Days/Fn::FindInMap
copilot/environments/addons/demodjango-s3-bucket.yml:116:21

W1011 FindInMap second key "RetentionDuration" doesn't exist in map "demodjangoS3BucketEnvironmentConfigMap" under "samm" at Resources/demodjangoS3BucketBucket/Properties/ObjectLockConfiguration/Fn::If/1/Rule/DefaultRetention/Fn::If/1/Days/Fn::FindInMap
copilot/environments/addons/demodjango-s3-bucket.yml:116:21

W1011 FindInMap second key "RetentionDuration" doesn't exist in map "demodjangoS3BucketEnvironmentConfigMap" under "gabe" at Resources/demodjangoS3BucketBucket/Properties/ObjectLockConfiguration/Fn::If/1/Rule/DefaultRetention/Fn::If/1/Days/Fn::FindInMap
copilot/environments/addons/demodjango-s3-bucket.yml:116:21

W1011 FindInMap second key "RetentionMode" doesn't exist in map "demodjangoS3BucketEnvironmentConfigMap" under "ant" at Resources/demodjangoS3BucketBucket/Properties/ObjectLockConfiguration/Fn::If/1/Rule/DefaultRetention/Fn::If/2/Mode/Fn::FindInMap
copilot/environments/addons/demodjango-s3-bucket.yml:117:85

W1011 FindInMap second key "RetentionMode" doesn't exist in map "demodjangoS3BucketEnvironmentConfigMap" under "toolspr" at Resources/demodjangoS3BucketBucket/Properties/ObjectLockConfiguration/Fn::If/1/Rule/DefaultRetention/Fn::If/2/Mode/Fn::FindInMap
copilot/environments/addons/demodjango-s3-bucket.yml:117:85

W1011 FindInMap second key "RetentionMode" doesn't exist in map "demodjangoS3BucketEnvironmentConfigMap" under "staging" at Resources/demodjangoS3BucketBucket/Properties/ObjectLockConfiguration/Fn::If/1/Rule/DefaultRetention/Fn::If/2/Mode/Fn::FindInMap
copilot/environments/addons/demodjango-s3-bucket.yml:117:85

W1011 FindInMap second key "RetentionMode" doesn't exist in map "demodjangoS3BucketEnvironmentConfigMap" under "jayesh" at Resources/demodjangoS3BucketBucket/Properties/ObjectLockConfiguration/Fn::If/1/Rule/DefaultRetention/Fn::If/2/Mode/Fn::FindInMap
copilot/environments/addons/demodjango-s3-bucket.yml:117:85

W1011 FindInMap second key "RetentionMode" doesn't exist in map "demodjangoS3BucketEnvironmentConfigMap" under "willg" at Resources/demodjangoS3BucketBucket/Properties/ObjectLockConfiguration/Fn::If/1/Rule/DefaultRetention/Fn::If/2/Mode/Fn::FindInMap
copilot/environments/addons/demodjango-s3-bucket.yml:117:85

W1011 FindInMap second key "RetentionMode" doesn't exist in map "demodjangoS3BucketEnvironmentConfigMap" under "dev" at Resources/demodjangoS3BucketBucket/Properties/ObjectLockConfiguration/Fn::If/1/Rule/DefaultRetention/Fn::If/2/Mode/Fn::FindInMap
copilot/environments/addons/demodjango-s3-bucket.yml:117:85

W1011 FindInMap second key "RetentionMode" doesn't exist in map "demodjangoS3BucketEnvironmentConfigMap" under "john" at Resources/demodjangoS3BucketBucket/Properties/ObjectLockConfiguration/Fn::If/1/Rule/DefaultRetention/Fn::If/2/Mode/Fn::FindInMap
copilot/environments/addons/demodjango-s3-bucket.yml:117:85

W1011 FindInMap second key "RetentionMode" doesn't exist in map "demodjangoS3BucketEnvironmentConfigMap" under "yusuf" at Resources/demodjangoS3BucketBucket/Properties/ObjectLockConfiguration/Fn::If/1/Rule/DefaultRetention/Fn::If/2/Mode/Fn::FindInMap
copilot/environments/addons/demodjango-s3-bucket.yml:117:85

W1011 FindInMap second key "RetentionMode" doesn't exist in map "demodjangoS3BucketEnvironmentConfigMap" under "samm" at Resources/demodjangoS3BucketBucket/Properties/ObjectLockConfiguration/Fn::If/1/Rule/DefaultRetention/Fn::If/2/Mode/Fn::FindInMap
copilot/environments/addons/demodjango-s3-bucket.yml:117:85

W1011 FindInMap second key "RetentionMode" doesn't exist in map "demodjangoS3BucketEnvironmentConfigMap" under "gabe" at Resources/demodjangoS3BucketBucket/Properties/ObjectLockConfiguration/Fn::If/1/Rule/DefaultRetention/Fn::If/2/Mode/Fn::FindInMap
copilot/environments/addons/demodjango-s3-bucket.yml:117:85

W1011 FindInMap second key "RetentionDuration" doesn't exist in map "demodjangoS3BucketEnvironmentConfigMap" under "ant" at Resources/demodjangoS3BucketBucket/Properties/ObjectLockConfiguration/Fn::If/1/Rule/DefaultRetention/Fn::If/2/Years/Fn::FindInMap
copilot/environments/addons/demodjango-s3-bucket.yml:121:21

W1011 FindInMap second key "RetentionDuration" doesn't exist in map "demodjangoS3BucketEnvironmentConfigMap" under "toolspr" at Resources/demodjangoS3BucketBucket/Properties/ObjectLockConfiguration/Fn::If/1/Rule/DefaultRetention/Fn::If/2/Years/Fn::FindInMap
copilot/environments/addons/demodjango-s3-bucket.yml:121:21

W1011 FindInMap second key "RetentionDuration" doesn't exist in map "demodjangoS3BucketEnvironmentConfigMap" under "staging" at Resources/demodjangoS3BucketBucket/Properties/ObjectLockConfiguration/Fn::If/1/Rule/DefaultRetention/Fn::If/2/Years/Fn::FindInMap
copilot/environments/addons/demodjango-s3-bucket.yml:121:21

W1011 FindInMap second key "RetentionDuration" doesn't exist in map "demodjangoS3BucketEnvironmentConfigMap" under "jayesh" at Resources/demodjangoS3BucketBucket/Properties/ObjectLockConfiguration/Fn::If/1/Rule/DefaultRetention/Fn::If/2/Years/Fn::FindInMap
copilot/environments/addons/demodjango-s3-bucket.yml:121:21

W1011 FindInMap second key "RetentionDuration" doesn't exist in map "demodjangoS3BucketEnvironmentConfigMap" under "willg" at Resources/demodjangoS3BucketBucket/Properties/ObjectLockConfiguration/Fn::If/1/Rule/DefaultRetention/Fn::If/2/Years/Fn::FindInMap
copilot/environments/addons/demodjango-s3-bucket.yml:121:21

W1011 FindInMap second key "RetentionDuration" doesn't exist in map "demodjangoS3BucketEnvironmentConfigMap" under "dev" at Resources/demodjangoS3BucketBucket/Properties/ObjectLockConfiguration/Fn::If/1/Rule/DefaultRetention/Fn::If/2/Years/Fn::FindInMap
copilot/environments/addons/demodjango-s3-bucket.yml:121:21

W1011 FindInMap second key "RetentionDuration" doesn't exist in map "demodjangoS3BucketEnvironmentConfigMap" under "john" at Resources/demodjangoS3BucketBucket/Properties/ObjectLockConfiguration/Fn::If/1/Rule/DefaultRetention/Fn::If/2/Years/Fn::FindInMap
copilot/environments/addons/demodjango-s3-bucket.yml:121:21

W1011 FindInMap second key "RetentionDuration" doesn't exist in map "demodjangoS3BucketEnvironmentConfigMap" under "yusuf" at Resources/demodjangoS3BucketBucket/Properties/ObjectLockConfiguration/Fn::If/1/Rule/DefaultRetention/Fn::If/2/Years/Fn::FindInMap
copilot/environments/addons/demodjango-s3-bucket.yml:121:21

W1011 FindInMap second key "RetentionDuration" doesn't exist in map "demodjangoS3BucketEnvironmentConfigMap" under "samm" at Resources/demodjangoS3BucketBucket/Properties/ObjectLockConfiguration/Fn::If/1/Rule/DefaultRetention/Fn::If/2/Years/Fn::FindInMap
copilot/environments/addons/demodjango-s3-bucket.yml:121:21

W1011 FindInMap second key "RetentionDuration" doesn't exist in map "demodjangoS3BucketEnvironmentConfigMap" under "gabe" at Resources/demodjangoS3BucketBucket/Properties/ObjectLockConfiguration/Fn::If/1/Rule/DefaultRetention/Fn::If/2/Years/Fn::FindInMap
copilot/environments/addons/demodjango-s3-bucket.yml:121:21
```

**Comment by kddejong** [comment]
@gabelton I might be missing something the Map `demodjangoS3BucketEnvironmentConfigMap` doesn't have the keys `RetentionDuration` or `RetentionDuration` ?  I don't see a part of `AWS::LanguageExtensions` that would fix that as there isn't a default specified for the `FindInMap`

**Comment by kddejong** [comment]
I think i get what you are saying though... Since the condition is also calling out the FindInMap.  But even if `demodjangoS3BucketRetentionPolicyDurationIsInDays` is true or false you are doing a FindInMap to `RetentionMode` or `RetentionDuration` which isn't in the map.  Wouldn't you need to add a default for those as well?

**Comment by kddejong** [comment]
Something like this...

```yaml
ObjectLockConfiguration:
        !If
        - demodjangoS3BucketApplyRetentionPolicy
        - ObjectLockEnabled: Enabled
          Rule:
            DefaultRetention: !If
              - demodjangoS3BucketRetentionPolicyDurationIsInDays
              - Mode: !FindInMap [demodjangoS3BucketEnvironmentConfigMap, !Ref Env, RetentionMode, {"DefaultValue": "COMPLIANCE"}]
                Days: !FindInMap
                  - demodjangoS3BucketEnvironmentConfigMap
                  - !Ref Env
                  - RetentionDuration
                  - DefaultValue: 90
              - Mode: !FindInMap [demodjangoS3BucketEnvironmentConfigMap, !Ref Env, RetentionMode, {"DefaultValue": "COMPLIANCE"}]
                Years: !FindInMap
                  - demodjangoS3BucketEnvironmentConfigMap
                  - !Ref Env
                  - RetentionDuration
                  - DefaultValue: 1
        - !Ref AWS::NoValue
```

**Comment by kddejong** [comment]
I'm doing a small tweak to help you out in v0.  This is resolved in v1.  V1 we have better context as we move down a template.

**Comment by gabelton** [comment]
> I think i get what you are saying though... Since the condition is also calling out the FindInMap. But even if `demodjangoS3BucketRetentionPolicyDurationIsInDays` is true or false you are doing a FindInMap to `RetentionMode` or `RetentionDuration` which isn't in the map. Wouldn't you need to add a default for those as well?

I think I already have defaults for both of those FindInMap though:

`- DefaultValue: No retention`

`- DefaultValue: Days`

**Comment by kddejong** [comment]
> > I think i get what you are saying though... Since the condition is also calling out the FindInMap. But even if `demodjangoS3BucketRetentionPolicyDurationIsInDays` is true or false you are doing a FindInMap to `RetentionMode` or `RetentionDuration` which isn't in the map. Wouldn't you need to add a default for those as well?
> 
> I think I already have defaults for both of those FindInMap though:
> 
> `- DefaultValue: No retention`
> 
> `- DefaultValue: Days`

You do where it matters in the Condition.  I missed that at first.
